common/djangoapps/rocket_chat/views.py

```diff
@@ -0,0 +1,93 @@
+
+from .utils import create_course_group, create_user, get_rocket_chat_settings
+
+
+@login_required
+def rocket_chat_discussion(request, course_id):
+
+    if not configuration_helpers.get_value('ENABLE_ROCKET_CHAT_SERVICE'):
+        raise Http404
+
+    course_key = CourseKey.from_string(course_id)
+
+    with modulestore().bulk_operations(course_key):
+
+        user = request.user
+        course = get_course_with_access(user, 'load', course_key)
+
+        staff_access = has_access(user, 'staff', course)
+        user_is_enrolled = CourseEnrollment.is_enrolled(user, course.id)
+
+        course_homepage_invert_title = configuration_helpers.get_value(
+            'COURSE_HOMEPAGE_INVERT_TITLE', False)
+
+        course_title = course.display_name_with_default
+        if course_homepage_invert_title:
+            course_title = course.display_number_with_default
+
+        context = {
+            'request': request,
+            'cache': None,
+            'course': course,
+            'course_title': course_title,
+            'staff_access': staff_access,
+            'user_is_enrolled': user_is_enrolled,
+        }
+
+        rocket_chat_settings = get_rocket_chat_settings()
+
+        if rocket_chat_settings:
+            api_rocket_chat = ApiRocketChat(rocket_chat_settings['admin_user'],
+                                            rocket_chat_settings['admin_pass'],
+                                            rocket_chat_settings['public_url_service']
+                                            )
+
+            user_info = api_rocket_chat.users_info(username=user.username)
+
+            try:
+                user_info = user_info.json()
+            except AttributeError:
+                create_user(api_rocket_chat, user, course_key)
+
+            if 'success' in user_info and not user_info['success']:
+                create_user(api_rocket_chat, user, course_key)
+
+            response = api_rocket_chat.users_create_token(
+                username=user.username)
+
+            try:
+                response = response.json()
+            except AttributeError:
+                context['rocket_chat_error_message'] = 'status_code = {}'.format(
+                    response.status_code)
+                return render_to_response('rocket_chat/rocket_chat.html', context)
+
+            if response['success']:
+                context['rocket_chat_data'] = response['data']
+                context['rocket_chat_url'] = rocket_chat_settings['public_url_service']
+                context['rocket_chat_error_message'] = None
+
+                create_course_group(api_rocket_chat, course_id, response['data']['userId'], user.username)
+
+            elif 'error' in response:
+                context['rocket_chat_error_message'] = response['error']
+
+            return render_to_response('rocket_chat/rocket_chat.html', context)
+
+        context['rocket_chat_error_message'] = 'RocketChat settings does not exist'
```

Comment:
Looking to this message, I prefer to send a different one in the context like "Rocket chat service is currently not available" or something, and use the message "RocketChat settings does not exist" in a log.

Reply:
@andrey-canon

Reply:
ok